### ISSUE TITLE
Incorrect liveness violation detection

### ISSUE THREAD

**Issue by db7** [MPI-SWS/genmc#17]
As far as I understand, this small example should not hang (no liveness violation).

```c
#include <assert.h>
#include <pthread.h>
#include <stdatomic.h>
#include <stdbool.h>

int x;
atomic_int lock = 1;

void *run_t0(void *arg) {
    /* initial lock owner */
    x = 1; // critical section 
   
    /* release lock (SC) */
    lock = 0; 
    return NULL;
}

static void *run_t1(void *arg) {
    /* acquire lock */
    do {
        while (lock != 0);
    } while (atomic_exchange(&lock, 1));

    /* because atomic_exchange is acquire, and lock = 0 is release, the
     * following assertion must be true */
    assert(x == 1 && "unexpected value");

    return NULL;
}

int main() {
    pthread_t t1;
    pthread_create(&t1, 0, run_t1, 0);
    run_t0(NULL);
    pthread_join(t1, NULL);
    return 0;
}
```

Running this with `-check-liveness` gives me the following error:
```
user@94ae0962b2ea:~/genmc-0.6$ genmc -mo -imm --check-liveness liveness-bug.c 
Error detected: Liveness violation!
Event (1, 3) in graph:
<-1, 0> main:
	(0, 0): B
	(0, 1): TC [forks 1] L.33
	(0, 2): Wna (x, 1) L.11
	(0, 3): Wsc (lock, 0) L.14
	(0, 4): TJ L.35
<0, 1> run_t1:
	(1, 0): B
	(1, 1): LOOP_BEGIN
	(1, 2): SPIN_START
	(1, 3): SPIN_START

Non-terminating spinloop
Number of complete executions explored: 0
Number of blocked executions seen: 1
Total wall-clock time: 0.02s
```

Could this be a caused because of the nested loops?

**Comment by michaliskok** [comment]
I have not investigated this thoroughly yet, but it seems that you are right: nested loops could be the cause of this.

Thanks a lot for the bug report!

**Comment by michaliskok** [comment]
Fixed in v0.6.1. Please see `genmc.h` for a slightly modified way on how to use `__VERIFIER_spin_{start,end}`. (Also adding a reference to #16.)

**Comment by db7** [comment]
Thanks for all the fixes! That is awesome! I'll start testing our code with version 0.6.1.
